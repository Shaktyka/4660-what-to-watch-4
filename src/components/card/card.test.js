import React from 'react';
import renderer from 'react-test-renderer';
import Card from './card.jsx';

const Film = {
  ID: 1,
  TITLE: `Mindhunter`,
  PREVIEW: `mindhunter.jpg`
};

describe(`Card rendering`, () => {

  it(`Card renders correctly`, () => {
    const tree = renderer
      .create(
          <Card
            id={Film.ID}
            title={Film.TITLE}
            poster={Film.PREVIEW}
            mainTitleClickHandler={() => {}}
            onHoverCard={() => {}}
          />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
