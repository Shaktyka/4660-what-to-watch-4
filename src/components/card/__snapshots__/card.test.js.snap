// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card rendering Card renders correctly 1`] = `
<article
  className="small-movie-card catalog__movies-card"
  id={7}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="mindhunter.jpg"
      type="video/webm"
      width="280"
    >
      Sorry, your browser doesn't support embedded video, but don't worry, you can 
      <a
        href="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      >
        download it
      </a>
      and watch it with your favorite video player!
    </video>
  </div>
  <h3
    className="small-movie-card__title"
    onClick={[Function]}
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Mindhunter
    </a>
  </h3>
</article>
`;
